lib/collection/property.js

```diff
@@ -180,9 +184,17 @@ _.extend(Property, /** @lends Property */ {
         // convert the variables to a substitutor object (will not reconvert if already substitutor)
         variables = Substitutor.box(variables, Substitutor.DEFAULT_VARS);
 
-        return _.merge(mutate ? obj : {}, obj, function (objectValue, sourceValue) {
-            return _.isString(sourceValue) ?  this.replaceSubstitutions(sourceValue, variables) : undefined;
-        }, this);
+        return _.mergeWith(mutate ? obj : {}, obj, function customizer(objectValue, sourceValue) {
+            if (!_.isString(sourceValue)) {
```

Comment:
Can you add a test case for this with a very deeply nested object containing a mixture of Arrays and other objects?